src/components/molecules/OakModal/OakModal.test.tsx

```diff
@@ -20,16 +20,26 @@ jest.mock("react-dom", () => {
 });
 
 describe(OakModal, () => {
-  it("matches snapshot", () => {
+  it("does not render until mounted on the client", () => {
     const tree = create(
       <OakThemeProvider theme={oakDefaultTheme}>
         <OakModal isOpen onClose={() => {}} footerSlot="Modal footer">
           Modal content
         </OakModal>
       </OakThemeProvider>,
-    ).toJSON();
+    );
+
+    expect(tree.toJSON()).toBeNull();
+  });
+
+  it("matches snapshot when mounted", async () => {
+    const result = renderWithTheme(
+      <OakModal isOpen onClose={() => {}}>
+        Modal content
+      </OakModal>,
+    );
 
-    expect(tree).toMatchSnapshot();
+    expect(result.container).toMatchSnapshot();
```

Comment:
Waiting for the shallow renderer to run hooks (which is necessary now that the component is not rendered immediately) introduced this error https://github.com/theKashey/react-focus-lock/issues/123 from `react-focus-lock` which I couldn't fix without extreme measures. 

The path of least resistance was to snapshot the DOM instead, which tbh I prefer.